commit f16e12d5e755d371247202fcccbcccd1ec16b2cf
Author: Jia Tan <jiat0218@gmail.com>
Date:   Wed Aug 17 20:20:16 2022 +0800

    liblzma: Add NULL check to lzma_index_hash_append.
    
    This is for consistency with lzma_index_append.

diff --git a/src/liblzma/common/index_hash.c b/src/liblzma/common/index_hash.c
index c3c56674..f55f7bc8 100644
--- a/src/liblzma/common/index_hash.c
+++ b/src/liblzma/common/index_hash.c
@@ -145,7 +145,7 @@ lzma_index_hash_append(lzma_index_hash *index_hash, lzma_vli unpadded_size,
 		lzma_vli uncompressed_size)
 {
 	// Validate the arguments.
-	if (index_hash->sequence != SEQ_BLOCK
+	if (index_hash == NULL || index_hash->sequence != SEQ_BLOCK
 			|| unpadded_size < UNPADDED_SIZE_MIN
 			|| unpadded_size > UNPADDED_SIZE_MAX
 			|| uncompressed_size > LZMA_VLI_MAX)